### ISSUE TITLE
object stereotypes and DataPortal

### ISSUE THREAD

**Issue by Chicagoan2016** [MarimerLLC/cslaforum#642]
**Question**
Hi All,
Is there any limitation on what Datportal methods can be called from a certain object stereotype?
RoleEditManager.cs in ProjectTracker has called DataPortal.CreateChild(), DataPortal.FetchChild() and DataPortal.Execute()

Kind regards
**Version and Platform**
CSLA version: 4.7.100
OS: Windows
Platform: WinForms, WPF, ASP.NET Core, MVC, Xamarin, etc.

**Comment by rockfordlhotka** [comment]
No, as long as you understand the ramifications of each type of call.

* Create/Fetch/Update/Delete will all route to the logical server
* *Child will all run locally in the context of your current code

**Comment by jonnybee** [comment]
And you should create CriteriaObjects (parameters object to root DataPortal methods) derived from CriteriaBase<> 

Also be aware that DataPortal.Execute() is special case of DataPortal.Update() over the wire.

**Comment by Chicagoan2016** [comment]
Thank you @rockfordlhotka  and @jonnybee 

@jonnybee , I never thought about DataPortal.Update()--- client side. Since Save() takes care of Inserts and updates.

**Comment by kcabral817** [comment]
_Also be aware that DataPortal.Execute() is special case of DataPortal.Update() over the wire._

I've been trying to understand why this statement invokes the update:
```
var obj3 = DataPortal.ExecuteAsync(obj2).Result;
```

**BO**
`[Csla.Server.ObjectFactory(typeof(SupplierFactory), "Create", "GetAsync", "UpdateAsync", "DeleteAsync", "ValidateAsync")]`
` public async Task<SupplierInfoPersist> ValidateAsync(SupplierInfoPersist info)
        {
            var item = await DataPortal.ExecuteAsync(info); 
            return item;
        }`

**FACTORY**
```
 public object ValidateAsync(SupplierInfoPersist obj)
        {
            CheckRules(obj);
            return obj;
        }
```

I don't understand my the checkrules never gets executed, but the update fires.  Any ideas?

**Comment by kcabral817** [comment]
I think part of the issue is getting the message out to the developers.  I heard about CSLA through Brian Lagunas.  Had he not mentioned it in passing, I would not have adopted it.  But just as importantly, we need to get into StackOverFlow for questions and answers to problems.  That is where a lot of developers go when asking questions.  Additionally, it's my belief we need representation at developer conferences, classes online, and support or mentoring on CodeMentor would not hurt.
